test/util.go

```diff
@@ -1,10 +1,53 @@
 package test
 
 import (
+	"fmt"
+	"image"
+	"image/draw"
+	"image/png"
+	"os"
+	"path/filepath"
+	"testing"
+
 	"fyne.io/fyne"
 	"fyne.io/fyne/internal/cache"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
+// AssertImagesEqual asserts that the master image at given filename (below testdata) and the given image are equal.
+// It fails the test if not. In this case the actual image is written to disk and reported.
+func AssertImagesEqual(t *testing.T, masterFilename string, got image.Image) {
+	wd, err := os.Getwd()
+	require.NoError(t, err)
+	masterPath := filepath.Join("testdata", masterFilename)
+	var expected image.Image
+	_, err = os.Stat(masterPath)
+	if os.IsNotExist(err) {
```

Comment:
If it does not exist how about writing the image out to disk and failing straigt off - possibly even erroring instead of failing, as that is really what is happening.

This would allow the "else" below to drop to the main body of the method where I think it belongs.
Perhaps the writing out of a failed image moves to a helper method and it can be called in this error case and the fail case below.